### PR TITLE
[Agency Dashboard] Dashboard UI scaffolding 1/n

### PR DESCRIPTION
**Pull request by terryttsai**
## Description of the change

Screen Shot 2022-11-11 at 1 37 12 PM

Implemented the new left side panel and all metrics button + header bar. Gave the buttons a hover state.

There's a tiny unfinished modal for the about page, that will be fleshed out in a later PR!

To follow up in later PRs:
- Rearrange the metric dropdowns + insights
- Add modals: 
  - About page 
  - Learn More
  - All Metrics
- Responsive layout
- Mobile filters view

## Type of change

> All pull requests must have at least one of the following labels applied (otherwise the PR will fail):

| Label                       	| Description                                                                                               	|
|-----------------------------	|-----------------------------------------------------------------------------------------------------------	|
| Type: Bug                   	| non-breaking change that fixes an issue                                                                   	|
| Type: Feature               	| non-breaking change that adds functionality                                                               	|
| Type: Breaking Change       	| fix or feature that would cause existing functionality to not work as expected                            	|
| Type: Non-breaking refactor 	| change addresses some tech debt item or prepares for a later change, but does not change functionality    	|
| Type: Configuration Change  	| adjusts configuration to achieve some end related to functionality, development, performance, or security 	|
| Type: Dependency Upgrade      | upgrades a project dependency - these changes are not included in release notes                             |

## Related issues

Closes #XXXX

### Development

**This box MUST be checked by the submitter prior to merging**:
- [ ] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: agency-dashboard/src/stores/DatapointsStore.ts [hunk 1]
```diff
@@ -133,6 +133,15 @@ class DatapointsStore extends BaseDatapointsStore {
       runInAction(() => {
         this.loading = false;
       });
+
+      const response2 = (await request({
```
Comment (lilidworkin) [inline agency-dashboard/src/stores/DatapointsStore.ts:137]: We actually don't need the published_datapoints endpoint anymore, do we? Want to remove the code here and also from the backend?

File: agency-dashboard/src/DashboardView.styles.tsx [hunk 2]
```diff
@@ -15,19 +15,191 @@
...
+
+  width: 25px;
+  height: 20px;
+  border-radius: 24px;
+
+  ${typography.sizeCSS.small}
+  font-size: 13px;
 `;
+
+export const AllMetricsButton = ({
```
Comment (lilidworkin) [inline agency-dashboard/src/DashboardView.styles.tsx:192]: nit, does this belong in a styles file?

File: agency-dashboard/src/DashboardView.styles.tsx [hunk 3]
```diff
@@ -15,19 +15,191 @@
...
+    padding-left: 0px;
+  }
+`;
+
+const MetricOverviewActionButtonText = styled.div`
+  ${typography.sizeCSS.normal}
+  margin-left: 8px;
+`;
+
+export const MetricOverviewActionShareButton = () => (
```
Comment (lilidworkin) [inline agency-dashboard/src/DashboardView.styles.tsx:124]: similar, do these belong in a styles file? not sure what best practice is

**Comment by terryttsai** [comment]
Sorry, I meant to open this as a draft PR! Videos will be coming soon. Just opening the PR for transparency, though I was planning on adding the modals before removing the WIP and asking for review

**Comment by terryttsai** [comment]
Though I was also thinking of keeping this as a small PR and adding the modals in a follow-up

File: agency-dashboard/src/stores/DatapointsStore.ts (hunk 1 above)
Comment (terryttsai) [reply agency-dashboard/src/stores/DatapointsStore.ts:137]: Yes sorry, the PR isn't ready for review yet so hadn't removed this yet!

File: agency-dashboard/src/DashboardView.styles.tsx (hunk 2 above)
Comment (terryttsai) [reply agency-dashboard/src/DashboardView.styles.tsx:192]: Good question, I think for now I'll leave these out of the styles file!

**Comment by lilidworkin** [comment]
> Sorry, I meant to open this as a draft PR! Videos will be coming soon. Just opening the PR for transparency, though I was planning on adding the modals before removing the WIP and asking for review

Got it! No worries, sorry for jumping the gun on review :)

> Though I was also thinking of keeping this as a small PR and adding the modals in a follow-up

That actually makes sense to me! You know how I love small / incremental PRs :)

**Comment by terryttsai** [comment]
In that spirit, I'm keeping this PR small and as-is, will be putting new work following up this PR!

**Review by lilidworkin — APPROVED** [review]
Love it! +1 for small/incremental PRs!

**Review by mxosman — APPROVED** [review]
Code LGTM, Terry! I wasn't able to run it for some reason - I ran `yarn install` and `yarn run dev` but the site doesn't load on localhost:3000 nor localhost:3001. Wondering if it's just me/something in my local env.

**Comment by mxosman** [comment]
> What's the error you're getting? does yarn dev compile successfully?

No errors at all actually - just the "This site can’t be reached. localhost refused to connect." screen.

**Comment by mxosman** [comment]
> > What's the error you're getting? does yarn dev compile successfully?
> 
> No errors at all actually - and everything compiles successfully - just the "This site can’t be reached. localhost refused to connect." screen.

**Comment by terryttsai** [comment]
Even across different browsers? I was having that issue on chrome until I realized I had network disabled in the chrome debugging tools...

**Pull request merged**